### PR TITLE
New cool and nice version of PCA Services - MongoDB

### PR DESCRIPTION
**Pull request by amouhoub**
In this PR, we (@Aminelouati &me) push a new and fresh refactor of PCA services (includes only MongoDB for the moment)
This refactored version is much more generalized and allows to effortlessly clone new services and states.
Enjoy !!!

### CODE REVIEW

File: DatabaseServices/resources/catalog/MongoDB_Service_Start.xml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+          onTaskError="cancelJob" >
+      <description>
+        <![CDATA[ Start the MongoDB server as a service. ]]>
+      </description>
+      <variables>
+        <variable name="SERVICE_ID" value="MongoDB" inherited="false" />
+      </variables>
+      <genericInformation>
+        <info name="task.icon" value="/automation-dashboard/styles/patterns/img/wf-icons/mongodb.png"/>
+        <info name="task.documentation" value="https://doc.activeeon.com/latest/user/ProActiveUserGuide.html#_nosql"/>
```
Comment (Aminelouati) [inline DatabaseServices/resources/catalog/MongoDB_Service_Start.xml]: documentation task -->https://doc.activeeon.com/latest/PCA/PCAUserGuide.html#_main_principles_of_pca

File: MongoDB/resources/catalog/MongoDB.xml [hunk 2]
```diff
@@ -118,69 +108,66 @@ println("--- END Loop_Over_Instance_Status ---")
...
       <scriptExecutable>
         <script>
           <code language="bash">
             <![CDATA[
-echo "--- BEGIN Start_MongoDB ---"
-echo "Pulling MongoDB image"
+echo BEGIN "$variables_PA_TASK_NAME"
+echo "Pulling "$variables_PA_JOB_NAME" image"
 
 docker pull activeeon/mongo
```
Comment (Aminelouati) [inline MongoDB/resources/catalog/MongoDB.xml]: For generalization concern, we need to change the image name to mongodb then, the command to 
docker pull activeeon/"$variables_PA_JOB_NAME" (lower case)

File: MongoDB/resources/catalog/Stop_MongoDB.xml [hunk 3]
```diff
@@ -27,30 +25,30 @@
       </description>
       <genericInformation>
         <info name="task.icon" value="/automation-dashboard/styles/patterns/img/wf-icons/mongodb.png"/>
+        <info name="Documentation" value="/automation-dashboard/styles/patterns/img/wf-icons/mysql.png"/>
```
Comment (luispinedamx) [inline MongoDB/resources/catalog/Stop_MongoDB.xml]: The value for documentation is not right, I think you copy-pasted the wrong path.

File: DatabaseServices/resources/catalog/MongoDB_Service_Start.xml (hunk 1 above)
Comment (amouhoub) [reply DatabaseServices/resources/catalog/MongoDB_Service_Start.xml]: Done

File: MongoDB/resources/catalog/Stop_MongoDB.xml (hunk 3 above)
Comment (amouhoub) [reply MongoDB/resources/catalog/Stop_MongoDB.xml]: I think you are right :D thx

File: MongoDB/resources/catalog/MongoDB.xml (hunk 2 above)
Comment (amouhoub) [reply MongoDB/resources/catalog/MongoDB.xml]: We'd better use a task variable for this

File: DatabaseServices/resources/catalog/MongoDB_Database_Interaction.xml [hunk 4]
```diff
@@ -508,17 +389,156 @@ print("END Import Data")
...
+        <script>
+          <code language="groovy">
+            <![CDATA[
+println("--- BEGIN PRE-SCRIPT " + variables.get("PA_TASK_NAME") + " ---")
+
+/*********************************************************************************
+* THIS PRESCRIPT MAPS THE SERVICE SPECIFIC VARIABLES WITH GENERIC VARIABLE NAMES *
+* THIS AVOIDS YOU FROM CHANGING THE WHOLE PCA WORKFLOWS IMPLEMENTATIONS          *
+*********************************************************************************/
+
```
[the remark this answers is not in the document]
Comment (amouhoub) [reply DatabaseServices/resources/catalog/MongoDB_Database_Interaction.xml]: In fact, we made the PCA services pretty generic in order to facilitate cloning them for creating new ones.
Therefore, here, we do a mapping from the example workflow in order to map MONGODB_ specific variables with pca variables...

**Pull request merged**